packages/fcl/src/utils/chain-id/chain-id-watcher.js

```diff
@@ -0,0 +1,21 @@
+import {config} from "@onflow/config"
+import {getChainId} from "./get-chain-id"
+
+/**
+ * @description
+ * Watches the config for changes to access node and updates the chain id accordingly
+ *
+ * @returns {Function} A function that unsubscribes the listener
+ *
+ */
+export function watchForChainIdChanges() {
+  let prevAccessNode
+  return config.subscribe(config => {
+    // Call getChainId to update the chainId cache if access node has changed
+    const newAccessNode = config?.["accessNode.api"]
+    if (prevAccessNode !== newAccessNode) {
+      getChainId().catch(() => {})
+    }
+    prevAccessNode = newAccessNode
```

Comment:
can't we avoid cache with: 

```js
    // Call getChainId to update the chainId cache if access node has changed
    const newAccessNode = config?.["accessNode.api"]
    if (prevAccessNode !== newAccessNode) {
      prevAccessNode = newAccessNode
      getChainId().catch(() => {
           prevAccessNode = null
      })
    }
    
```

Reply:
So the catch statement in this code is basically because the top level chain id caching has no guarantee to actually successfully update the cache.  If it fails, then we simply aim to fetch the value next time `getChainId` is called from somewhere else.  This catch statement alone is the majority of what fixes the existing issues that were being seen.  The rest of the caching behaviour is mostly preventative for certain race conditions I can foresee becoming a problem that haven't been realized yet.